src/lib/functions.mli

```diff
@@ -16,4 +16,9 @@ val erfc : float -> float
 val chi_square_less : float -> int -> float
 val chi_square_greater : float -> int -> float
 val t_lookup : float -> int -> float
-val softmax : ?temp:float -> float array -> int -> float
+
+(** [softmax ?temp weights] transform [weights] into softmax weights dependent
+    on [temp]erature.
+
+    @raise Invalid_argument if [weights] is empty or [temp = 0]. *)
+val softmax : ?temp:float -> float array -> float array
```

Comment:
I'd put `?temperature`, `?temp` out of context would be more “temporary,” and that kind of abbreviation is not worth it.